### PR TITLE
Update AutoMapper to 6.0.2

### PR DESCRIPTION
**Pull request by cormacpayne**
## Description

Initial attempt at fixing issue https://github.com/Azure/azure-powershell/issues/4294 by bumping the version of `AutoMapper` used and adopting a singleton pattern with the new version that will prevent the mapper from being populated every time a cmdlet is executed.

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md).**
- [ ] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/preview/documentation/help-generation.md).**
- [ ] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/preview/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.
- [ ] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [ ] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#supportsshouldprocess).
- [ ] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [ ] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [ ] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [ ] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

**Comment by cormacpayne** [comment]
Building packages job: http://azuresdkci.cloudapp.net/job/PrPack/18/

**Comment by cormacpayne** [comment]
Building packages job: http://azuresdkci.cloudapp.net/job/PrPack/19/

**Comment by cormacpayne** [comment]
Building packages job: http://azuresdkci.cloudapp.net/job/PrPack/20/

**Comment by cormacpayne** [comment]
Building packages job: http://azuresdkci.cloudapp.net/job/posh-pack/3/

File: setup/azurecmdfiles.wxi [hunk 1]
```diff
@@ -132,9 +132,6 @@
                         <Component Id="cmpBEA6E3263480C18705DC2E3B15E3C1CF" Guid="*">
                             <File Id="fil7E95F218EFC39408F9F2DB3FD01A124C" KeyPath="yes" Source="$(var.sourceDir)\ResourceManager\AzureResourceManager\AzureRM.ApiManagement\AutoMapper.dll" />
                         </Component>
-                        <Component Id="cmp957D146263F021CE3EF1CBADBF06D2C4" Guid="*">
```
Comment (markcowl) [inline setup/azurecmdfiles.wxi]: So we don't need this file anymore?

File: src/ResourceManager/RecoveryServices.SiteRecovery/Commands.RecoveryServices.SiteRecovery/Commands.RecoveryServices.SiteRecovery.csproj [hunk 2]
```diff
@@ -35,11 +35,12 @@
...
   <ItemGroup>
-    <Reference Include="AutoMapper, Version=3.1.1.0, Culture=neutral, PublicKeyToken=be96cd2c38ef1005, processorArchitecture=MSIL">
-      <HintPath>..\..\..\packages\AutoMapper.3.1.1\lib\net40\AutoMapper.dll</HintPath>
+    <Reference Include="AutoMapper, Version=6.0.2.0, Culture=neutral, PublicKeyToken=be96cd2c38ef1005, processorArchitecture=MSIL">
+      <HintPath>..\..\..\packages\AutoMapper.6.0.2\lib\net45\AutoMapper.dll</HintPath>
+      <Private>True</Private>
     </Reference>
-    <Reference Include="AutoMapper.Net4, Version=3.1.1.0, Culture=neutral, PublicKeyToken=be96cd2c38ef1005, processorArchitecture=MSIL">
-      <HintPath>..\..\..\packages\AutoMapper.3.1.1\lib\net40\AutoMapper.Net4.dll</HintPath>
+    <Reference Include="AutoMapper.Net4, Version=6.0.2.0, Culture=neutral, PublicKeyToken=be96cd2c38ef1005, processorArchitecture=MSIL">
```
Comment (markcowl) [inline src/ResourceManager/RecoveryServices.SiteRecovery/Commands.RecoveryServices.SiteRecovery/Commands.RecoveryServices.SiteRecovery.csproj]: This looks like AutoMapper.Net4 is still used in some circumstances

**Comment by cormacpayne** [comment]
@blueww Hey Wei, the bulk of the Storage code changed is in the [`ServiceManagementProfile`](https://github.com/Azure/azure-powershell/pull/4679/files#diff-90282ac36665778677b6f252402e6adcR449) class, which defines the mappings for types used in RDFE cmdlets. It appears the issue that the user is seeing in #4418 comes from the code [here](https://github.com/Azure/azure-powershell/blob/preview/src/ServiceManagement/Compute/Commands.ServiceManagement/StorageServices/GetAzureStorageAccount.cs#L52-L53) - AutoMapper is trying to map an object of type `StorageAccountGetResponse` to `StorageServicePropertiesOperationContext`, but it cannot find a corresponding mapping (even though it's defined in the `ServiceManagementProfile` class).

Currently, AutoMapper is being used statically throughout the modules, which is why we believe these issues with "unsupported mappings" are popping up - the correct mappings may not have been instantiated by the time the mapper is used. This PR updates the version of AutoMapper used to 6.0.2, which deprecates the support of a static mapper, and we define a specific mapper to be used for each module. This ensures that every module has their own mapper, every call to `Mapper.Map<Foo, Bar>()` is using _the module's specific mapper_, and that the mapper is instantiated with the correct mappings whenever it is used. 

Now, rather than calling `Mapper.Map<Foo,Bar>()` in the previously cited `GetAzureStorageAccount` class, you will now use the _specific mapper_ that has defined the mappings you will need, [`ServiceManagementProfile.Mapper.Map<Foo,Bar>()`](https://github.com/Azure/azure-powershell/pull/4679/files#diff-73b1dfab21cc91b15ec8b63eefc4695dR56).

**Comment by cormacpayne** [comment]
On-demand: http://azuresdkci.cloudapp.net/job/powershell-demand/1914/

**Comment by cormacpayne** [comment]
On-demand: http://azuresdkci.cloudapp.net/job/powershell-demand/1917/

**Comment by cormacpayne** [comment]
@azuresdkci test this please

**Comment by cormacpayne** [comment]
@azuresdkci test this please

**Comment by cormacpayne** [comment]
Received sign-off from Compute and Network that this does not break existing scenarios.

**Pull request merged**